src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -213,7 +213,7 @@ const ProgressiveProfiling = (props) => {
       {props.shouldRedirect ? (
         <RedirectLogistration
           success
-          redirectUrl={registrationResult.redirectUrl}
+          redirectUrl={queryParams?.next ? queryParams.next : registrationResult.redirectUrl}
```

Comment:
instead of using Recommendation's redirect logistration can we do something like

```
if (queryParams?.next) {
  <RedirectLogistration
          success
          redirectUrl={queryParams?.next ? queryParams.next : registrationResult.redirectUrl}
        />
} else {
    <RedirectLogistration
          success
          redirectUrl={registrationResult.redirectUrl}
          redirectToRecommendationsPage={showRecommendationsPage}
          educationLevel={values?.level_of_education}
          userId={authenticatedUser?.userId}
        />
}
```

this will separate the redirection logic for both. otherwise its getting mixup and will be hard to debug later.

Reply:
I think we are not writing any complex logic for this redirection here.
We can simplify this like `redirectUrl={queryParams?.next || registrationResult.redirectUrl} `